OpenRA.Game/Network/ChatLine.cs

```diff
@@ -0,0 +1,55 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using OpenRA.Primitives;
+
+namespace OpenRA.Network
+{
+	public enum ChatPool { System, Chat, Mission, Feedback }
+	public class ChatLine : IEquatable<ChatLine>
+	{
+		public readonly Color NameColor;
+		public readonly string Name;
+		public readonly string Text;
+		public readonly Color TextColor;
+		public readonly ChatPool Pool;
+
+		public ChatLine(string name, Color nameColor, string text, Color textColor, ChatPool chatPool)
+		{
+			NameColor = nameColor;
+			Name = name;
+			Text = text;
+			TextColor = textColor;
+			Pool = chatPool;
+		}
+
+		public bool IsRepeatable()
```

Comment:
IMO its not clear what "repeatable" actually means. How about making this a `RemoveOnDuplicate` (or similar, if my interepretation is wrong) property instead?

Reply:
Well, it's not just removing on duplicate because there's a counter being incremented and appended to the message (e.g. "Unit lost (5)"). But I agree that `IsRepeatable` is not be the best name.

Reply:
Renamed to `IncrementOnDuplicate`.